### PR TITLE
add message component, chat styles

### PR DESCRIPTION
**Pull request by Ponomaleks**

### CODE REVIEW

File: src/app/main/components/chat/chat.component.ts [hunk 1]
```diff
@@ -1,28 +1,89 @@
...
+    if (this.message.value) {
+      // send message
+      this.message.reset();
+      this.focusOnInput();
+    }
+  }
+
+  focusOnInput(): void {
+    this.messageInputRef.epmInputRef.nativeElement.focus();
+  }
```
Comment (GennadiyShpak) [inline src/app/main/components/chat/chat.component.ts:79]: why public?

File: src/app/shared/pipes/chat-start-day.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    const today = new Date();
+    if (value) {
+      const valueDate = new Date(value);
+      if (getISODateWithoutTime(valueDate) === getISODateWithoutTime(today)) {
+        return 'Today';
+      } else return super.transform(value, args);
+    }
+    return '';
+  }
+}
```
Comment (GennadiyShpak) [inline src/app/shared/pipes/chat-start-day.pipe.ts:22]: need to discuss

File: src/app/shared/pipes/hide-message-time.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+})
+export class HideMessageTimePipe implements PipeTransform {
+  transform(message: Message, nextMessage: Message | undefined): boolean {
+    if (!nextMessage || getISODateWithoutTime(message.time) !== getISODateWithoutTime(nextMessage.time)) {
+      return false;
+    }
+
+    return message.sender === nextMessage.sender;
+  }
+}
```
Comment (GennadiyShpak) [inline src/app/shared/pipes/hide-message-time.pipe.ts:17]: need to discuss

File: src/app/main/components/chat/chat.component.ts (hunk 1 above)
Comment (Ponomaleks) [reply src/app/main/components/chat/chat.component.ts:79]: fixed

File: src/app/shared/pipes/chat-start-day.pipe.ts (hunk 2 above)
Comment (Ponomaleks) [reply src/app/shared/pipes/chat-start-day.pipe.ts]: getISODateWithoutTime has been chenged to new DatePipe('en-US');

File: src/app/shared/pipes/hide-message-time.pipe.ts (hunk 3 above)
Comment (Ponomaleks) [reply src/app/shared/pipes/hide-message-time.pipe.ts]: getISODateWithoutTime has been chenged to new DatePipe('en-US');

**Review by GennadiyShpak — APPROVED** [review]

**Pull request merged**